app/controllers/posts_controller.rb

```diff
@@ -99,14 +92,14 @@ def fetch_post(viewed_by_seller = true)
       end
       e404 if @post.blank?
 
-      if viewed_by_seller
-        e404 if @post.seller != current_seller
+      if @post.seller_id?
+        @seller = @post.seller
+      else
+        @seller = @post.link.seller
       end
 
-      if @post.seller_id?
-        e404 if @post.seller.suspended?
-      elsif @post.link_id?
-        e404 if @post.link.seller&.suspended?
+      if viewed_by_seller
+        e404 if @seller != current_seller
```

Comment:
Setting `@seller` first as to use it in `ensure_seller_is_eligible_to_send_emails` down below